components/brave_wallet/browser/keyring_service.cc

```diff
@@ -1286,13 +1306,19 @@ absl::optional<std::string> KeyringService::AddAccountForKeyring(
   auto* keyring = GetHDKeyringById(keyring_id);
   if (!keyring)
     return absl::nullopt;
-  keyring->AddAccounts(1);
-  size_t accounts_num = keyring->GetAccountsNumber();
-  CHECK(accounts_num);
-  SetAccountMetaForKeyring(
-      profile_prefs_, GetAccountPathByIndex(accounts_num - 1, keyring_id),
-      account_name, keyring->GetAddress(accounts_num - 1), keyring_id);
-  return keyring->GetAccounts().at(accounts_num - 1);
+
+  auto added_accounts = keyring->AddAccounts(1);
+  if (added_accounts.empty()) {
+    return absl::nullopt;
+  }
+  DCHECK_EQ(added_accounts.size(), 1u);
+
+  SetDerivedAccountInfoForKeyring(
+      profile_prefs_,
+      DerivedAccountInfo(added_accounts[0].path, account_name,
+                         added_accounts[0].address),
+      keyring_id);
+  return added_accounts[0].address;
```

Comment:
Go into keyring once (AddAccounts) which returns all required data remember account in prefs.